cudamapper/src/index_generator_gpu.hpp

```diff
@@ -0,0 +1,61 @@
+/*
+* Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+*
+* NVIDIA CORPORATION and its licensors retain all intellectual property
+* and proprietary rights in and to this software, related documentation
+* and any modifications thereto.  Any use, reproduction, disclosure or
+* distribution of this software and related documentation without an express
+* license agreement from NVIDIA CORPORATION is strictly prohibited.
+*/
+
+#pragma once
+
+#include <cstdint>
+#include "cudamapper/index_generator.hpp"
+#include "cudamapper/sequence.hpp"
+#include "cudamapper/types.hpp"
+#include "minimizer.hpp"
+
+namespace claragenomics {
+    /// IndexGeneratorGPU - generates data structures necessary for building the actualy index using (k,w)-kmer minimizers
+    class IndexGeneratorGPU : public IndexGenerator {
+    public:
+        /// \brief generates data structures necessary for building the actualy index
+        ///
+        /// \param query_filename filepath to reads in FASTA or FASTQ format
+        /// \param minimizer_size k - the kmer length used as a minimizer
+        /// \param window_size w - the length of the sliding window used to find minimizer
+        IndexGeneratorGPU(const std::string& query_filename, std::uint64_t minimizer_size, std::uint64_t window_size);
+
+        /// \brief return minimizer size (k)
+        /// \return minimizer size
+        std::uint64_t minimizer_size() const;
```

Comment:
just curious if the range for `uint64_t` is needed for the minimizers. we've been using signed integers as a standard, an using https://github.com/clara-genomics/ClaraGenomicsAnalysis/blob/dev-v0.2.0/common/utils/include/claragenomics/utils/signed_integer_utils.hpp#L31 to check for negative values (such as users passing negative inputs). This is just to protect from overflow situations

Reply:
Ok, we'll fix that in a separate PR.
@vellamike FYI